File: OCTOBER 24/09 October/921. Minimum Add to Make Parentheses Valid LC.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        stack<char> stk; 
        int open = 0, close=0;
        for (char ch:s) {
            if(ch == '(')
                open++;
            else {
                if(open)
                    open--;
                else
                    close++;
            }
        }
        return open+close;
    }
};

File: APRIL 2025/05 April/AMZN/921. Minimum Add to Make Parentheses Valid LC.java
class Solution {
    public int minAddToMakeValid(String s) {
        int minAdd = 0, balance = 0;
        for(char ch:s.toCharArray()) {
            if(ch=='(')
                balance++;
            else
            {
                balance--;
                if(balance<0)
                {
                    minAdd++;
                    balance = 0;
                }
            }
        }
        minAdd += balance;
        return minAdd;
    }
}
